src/meta_type.rs

```diff
@@ -96,6 +96,16 @@ impl MetaType {
         }
     }
 
+    /// Creates a new meta type from the user supplied type id and type info function.
+    ///
+    /// NOTE: It is the responsibility of the caller to ensure unique type ids per custom type.
+    pub fn new_custom(type_id: u64, fn_type_info: fn() -> Type<MetaForm>) -> Self {
```

Comment:
Can `fn_type_info` be opened up to accept parameters? Without the ability to inject any dependencies into this function, It is actually not of much use if we want to use this in a really dynamic fashion...

Reply:
See #166

Reply:
I don't think this will work.